packages/api/app/auth.js

```diff
@@ -0,0 +1,65 @@
+
+'use strict';
+
+const {
+  JsonWebTokenError,
+  TokenExpiredError
+} = require('jsonwebtoken');
+const { User, AccessToken } = require('../models');
+const { verifyJwtToken } = require('../lib/token');
+
+/**
+ * An express middleware that checks if an incoming express
+ * request is authenticated
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @param {Function} next - express middleware callback function
+ * @returns {Promise<Object>} - promise of an express response object
+ */
+async function ensureAuthenticated(req, res, next) {
```

Comment:
Nitpick: I think authorization is a better term for what we're checking here than authentication

Reply:
3d483a83b